src/System.Data.Common/tests/System/Data/SqlTypes/SqlDecimalTest.cs

```diff
@@ -276,6 +276,52 @@ public void AdjustScale()
         }
 
         [Fact]
+        public void AdjustScaleForFlatZero()
```

Comment:
This method tests : 
* A case of when the scale of ***plain zero 0*** is adjusted by n (n >= 0) 
* A case of when a scale of zero with fraction (e.g. "0.00") is adjusted by n (n < 0) to see if it lands on ***plain zero 0*** with precision 1 & scale 0 correctly.

It is all about ***plain zero 0***.